Humans_Engine/Query/CvalueAnalysis.cs

```diff
@@ -82,8 +82,10 @@ private static List<Cvalue> EvaluateCvalue(Audience audience, CvalueSettings set
             if (audience.Spectators.Count == 0)
                 return results;
 
-            KDTree<Spectator> spectatorTree = SetKDTree(audience);
+            SetKDTree(audience);
```

Comment:
I've not yet tested, but wanted to get a couple of comments noted. I'm really liking the use of KD Trees here and think there's a case for broader use throughout the BHoM, specifically space planning at a multitude of scales. I do think this warrants at least another issue raised to introduce this functionality a bit more formally.

Reply:
@michaelhoehn noted and will raise an issue. Do you have a specific use case / project that would benefit from a more generic KD Tree. Also see the user review from Joe. This one is now ready for your scrutiny 😄

Reply:
Hey @rolyhudson I'm thinking about partitioning floorplates based on predefined constraints. For example, developers will frequently have a requirement to fit _n_ 1BR at 300sf, _n_ 2BR at 800sf, and so on. I imagine KD trees can be used to quickly partition a given space. Pair this with some controlling parameters (a seed for an easy example) and you can quickly output some layouts.